modules/apps/redirect/redirect-test/src/testIntegration/java/com/liferay/redirect/web/internal/instance/lifecycle/test/RedirectEntryPermissionPortalInstanceLifecycleListenerTest.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.redirect.web.internal.instance.lifecycle.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.model.Company;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.ResourceConstants;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.security.permission.ActionKeys;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ResourcePermissionLocalService;
+import com.liferay.portal.kernel.service.RoleLocalService;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.CompanyTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.redirect.model.RedirectEntry;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author tototrinh
+ */
+@RunWith(Arquillian.class)
+public class RedirectEntryPermissionPortalInstanceLifecycleListenerTest {
```

Comment:
@tototrinh To verify that the upgrade process is indeed not needed, you'll need to add an additional test:
1. Disable the instance initializer you created.
2. Add a new company.
3. Enable the instance initializer component.
4. Check that the company you created in 2 has the correct permissions.

To see how to do that, take a look at the AMThumbnailsOSGiCommandsTest. There the ServiceComponentRuntime is used to disable/enable components programatically.

If the test doesn't pass, you'll need to create an upgrade process for existing companies.

Reply:
Hi @adolfopa,
Thanks for your feedback. I update the code for test integration. Please help review it. Thanks.